API/src/routes/events.js

```diff
@@ -57,6 +59,40 @@ events.get('/:id', (req, res) => {
 	  });
 });
 
+/*token verification (put in here instead of index.js as the '/games' path had to be split
+  based on the verb being used, rather than just the path.
+  The same code is copied in games.
+*/
+events.use((req,res,next) => {
+  var token = req.body.token || req.query.token || req.headers['x-access-token'];
+
+if (token) {
+
+  jwt.verify(token, secret.secret, function(err, decoded) {
+
+    if (err) {return res.status(403)
+      .json({
+        message: "wrong token"
+      })
+    }
+
+    else {
+      //save decoded token for use elsewhere
+      req.decoded = decoded;
+      next();
+    }
+
+
+  });
+} else {
+  return res.status(403)
+    .json({
+      message: "no token"
+    });
+}
+});
+
```

Comment:
Sorry, can you briefly explain what this part is doing? :D sorry I don't get it. What's event.use?

Reply:
Events.use checks the token that the client gives you, and ensures its the not expired, and the same one that we gave them. Any type of task that requires authentication (ie. updating events) must be put below events.use. This ensures that all those tasks will go through events.use and hence require authentication in order for it to work.